nixos/modules/services/monitoring/watchdogd.nix

```
@@ -0,0 +1,128 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.watchdogd;
+
+  mkPluginOpts = plugin: {
+    enable = mkEnableOption "watchdogd plugin ${plugin}";
+    interval = mkOption {
+      type = types.int;
+      default = 300;
+      description = ''
+        The poll interval. Defaults to 300 seconds.
+      '';
+    };
+    logmark = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Log current stats every poll interval.
+        Defaults to disabled.
+      '';
+    };
+    warning = mkOption {
+      type = types.int;
+      description = ''
+        The high watermark level. Alert sent to log.
+      '';
+    };
+    critical = mkOption {
+      type = types.int;
```

I must say I'm not entirely sure what these options are, but I'm assuming they're not supposed to be negative numbers

```suggestion
      type = types.ints.unsigned;
      description = ''
        The high watermark level. Alert sent to log.
      '';
    };
    critical = mkOption {
      type = types.ints.unsigned;
```